packages/contracts-bedrock/src/L1/DataAvailabilityChallenge.sol

```diff
@@ -0,0 +1,419 @@
+/// @custom:field status The status of the challenge.
+/// @custom:field challenger The address that initiated the challenge.
+/// @custom:field startBlock The block number at which the challenge was initiated.
+struct Challenge {
+    address challenger;
+    uint256 lockedBond;
+    uint256 startBlock;
+    uint256 resolvedBlock;
+}
+
+/// @title DataAvailabilityChallenge
+/// @notice This contract enables data availability of a data commitment at a given block number to be challenged.
+///         To challenge a commitment, the challenger must first post a bond (bondSize).
+///         Challenging a commitment is only possible within a certain block interval (challengeWindow) after the
+///         commitment was made.
+///         If the challenge is not resolved within a certain block interval (resolveWindow), the challenge can be
+///         expired.
+///         If a challenge is expired, the challenger's bond is unlocked and the challenged commitment is added to the
+///         chain of expired challenges.
+contract DataAvailabilityChallenge is OwnableUpgradeable, ISemver {
+    /// @notice Error for when the provided resolver refund percentage exceeds 100%.
+    error InvalidResolverRefundPercentage(uint256 invalidResolverRefundPercentage);
+
+    /// @notice Error for when the challenger's bond is too low.
+    error BondTooLow(uint256 balance, uint256 required);
+
+    /// @notice Error for when attempting to challenge a commitment that already has a challenge.
+    error ChallengeExists();
+
+    /// @notice Error for when attempting to resolve a challenge that is not active.
+    error ChallengeNotActive();
+
+    /// @notice Error for when attempting to unlock a bond from a challenge that is not expired.
+    error ChallengeNotExpired();
+
+    /// @notice Error for when attempting to challenge a commitment that is not in the challenge window.
+    error ChallengeWindowNotOpen();
+
+    /// @notice Error for when the provided input data doesn't match the commitment.
+    error InvalidInputData(bytes providedDataCommitment, bytes expectedCommitment);
+
+    /// @notice Error for when the call to withdraw a bond failed.
+    error WithdrawalFailed();
+
+    /// @notice Error for when a the type of a given commitment is unknown
+    error UnknownCommitmentType(bytes1 commitmentType);
+
+    /// @notice Error for when the commitment length does not match the commitment type
+    error InvalidCommitmentLength(uint8 commitmentType, uint256 expectedLength, uint256 actualLength);
+
+    /// @notice An event that is emitted when the status of a challenge changes.
+    /// @param challengedCommitment The commitment that is being challenged.
+    /// @param challengedBlockNumber The block number at which the commitment was made.
+    /// @param status The new status of the challenge.
+    event ChallengeStatusChanged(
+        bytes indexed challengedCommitment, uint256 indexed challengedBlockNumber, ChallengeStatus status
```

Comment:
Are you sure that you want this to be indexed? If its larger than 31 bytes then the hash will be the value

Reply:
Good catch, this is a leftover from when the commitment was `bytes32`. Will remove the index.